packages/notifi-react-card/lib/hooks/useNotificationHistory.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  AccountBalanceChangedEventDetails,
+  BroadcastMessageEventDetails,
+  ChatMessageReceivedEventDetails,
+  GenericEventDetails,
+  HealthValueOverThresholdEventDetails,
+  NotificationHistoryEntry,
+} from '@notifi-network/notifi-core';
+import React from 'react';
+
+import { AnnouncementIcon } from '../assets/AnnouncementIcon';
+import { ChatAlertIcon } from '../assets/ChatAlertIcon';
+import { RatioCheckIcon } from '../assets/RatioCheckIcon';
+import { SwapIcon } from '../assets/SwapIcon';
+import { AlertIcon } from '../components/AlertHistory/AlertIcon';
+import { AlertNotificationViewProps } from '../components/AlertHistory/AlertNotificationRow';
+import { formatAmount } from '../utils/AlertHistoryUtils';
+
+export const useNotificationHistory = () => {
+  class SupportedEntryDetail {
```

Comment:
what's the benefit to this having a constructor? I think just an interface or a `type` declaration is sufficient